vertical-pod-autoscaler/pkg/utils/vpa/capping.go

```diff
@@ -153,6 +156,30 @@ func applyVPAPolicy(recommendation apiv1.ResourceList, policy *vpa_types.Contain
 	return annotations
 }
 
+// ApplyVPAPolicy returns a recommendation, adjusted to obey policy.
+func ApplyVPAPolicy(podRecommendation *vpa_types.RecommendedPodResources,
+	policy *vpa_types.PodResourcePolicy) (*vpa_types.RecommendedPodResources, error) {
+	if podRecommendation == nil {
+		return nil, nil
+	}
+	if policy == nil {
+		return podRecommendation, nil
+	}
+
+	updatedRecommendations := []vpa_types.RecommendedContainerResources{}
+	for _, containerRecommendation := range podRecommendation.ContainerRecommendations {
+		containerName := containerRecommendation.ContainerName
+		updatedContainerResources, _, err := getCappedRecommendationForContainer(
+			containerName, nil, &containerRecommendation, policy, false /* capToLimit */)
+
+		if err != nil {
```

Comment:
This currently cannot happen which might be treated as an argument to have a specialised function instead of resuing getCapped[...]

Reply:
Went for not reusing.